driver/src/orderbook/streamed/state.rs

```diff
@@ -23,6 +23,18 @@ pub struct State {
     balances: HashMap<(UserId, TokenAddress), Balance>,
     tokens: Tokens,
     last_solution: LastSolution,
+    /// True when we have received some trades or trade reverts but not yet the final solution
+    /// submission that indiciates all trades have been received.
+    solution_partially_received: bool,
```

Comment:
Maybe this fits better directly on the LastSolution struct or could be expressed by making last_solution an option that is None when it's in the process of receiving (although I'm not sure that's very explicit).

Reply:
This concerns me as it means there are times where the orderbook is unqueryable for future batches.

Reply:
Maybe the correct approach is to not include proceeds if a solution is pratially applied instead of erroring?

Reply:
I see your point. Since that is getting less straightforward and we are not actually going to solve optimistically yet I think we should restrict this to `Batch::Current` and revisit this later.

Reply:
Hmm but we do need the batch id in case the it has increased but we just didn't get an event for the next batch yet. Then I would still say we don't fix the optimistic solving case yet because it is more complicated but keep the code as is until we need optimistic solving or the event based orderbook has been merged and we have more time.

Reply:
Do you mind opening an issue so we don't forget?